python/ray/data/_internal/execution/operators/base_physical_operator.py

```diff
@@ -79,6 +80,13 @@ def num_outputs_total(self) -> Optional[int]:
             else self.input_dependencies[0].num_outputs_total()
         )
 
+    def num_output_rows_total(self) -> Optional[int]:
+        return (
+            self._output_rows
+            if self._output_rows
+            else self.input_dependencies[0].num_outputs_total()
```

Comment:
In the latest screenshot, for the `Shuffle Map` and `Shuffle Reduce` progress bars, the unit looks like it may still be in `bundles` rather than `rows`, since the count is much lower (197 vs 10k rows in the dataset). I think the current code gets the number of bundles rather than rows, so we can swap it out with the new method you added.

```suggestion
            else self.input_dependencies[0].num_output_rows_total()
```